internal/exec/stages/files/units.go

```diff
@@ -64,10 +92,69 @@ func (s *stage) createUnits(config types.Config) error {
 			s.relabel(relabelpath)
 		}
 	}
-	// and relabel the preset file itself if we enabled/disabled something
-	if enabledOneUnit {
-		s.relabel(util.PresetPath)
+	// if we have presets then create the systemd preset file.
+	if len(presets) != 0 {
+		if err := s.createSystemdPresetFile(presets); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// parseInstanceUnit is a helper function to extract the name
+// and coresponding instances for a given instantiated unit.
```

Comment:
`corresponding`.  Also, it'll extract exactly one instance, correct?

Reply:
Yeah. I think this `godoc` is a bit confusing. Let me change the wording to make it more clear. It should be `s/instances/instance`